merlin/core/dispatch.py

```diff
@@ -359,12 +359,12 @@ def concat_columns(args: list):
         return args[0]
     elif cudf is not None and isinstance(args[0], cudf.DataFrame):
         return cudf.concat(
-            [a.reset_index(drop=True) for a in args],
+            [a.reset_index(drop=True) for a in args if not a.empty],
             axis=1,
         )
     elif isinstance(args[0], pd.DataFrame):
         return pd.concat(
-            [a.reset_index(drop=True) for a in args],
+            [a.reset_index(drop=True) for a in args if not a.empty],
```

Comment:
From the NVT test failures, it looks like this `concat` call fails if there are no dataframes to concatenate, so the list comprehension might need to move out to a local that can be checked before calling `concat`

Reply:
I've updated with this logic, and still have an error. It seems ignoring empty dataframes is maybe not the correct thing to do in the general case. NVTabular is relying on concating empty dataframes where the index is preserved (concated).